backend/service-question/src/main/java/g55/cs3219/backend/questionservice/service/QuestionService.java

```diff
@@ -54,39 +57,34 @@ public List<Question> getQuestionsByFilters(String category, String difficulty)
         }
     }
 
-    public List<Question> getQuestionsByDifficulty(String difficulty) {
+    public List<QuestionDto> getQuestionsByDifficulty(String difficulty) {
         List<Question> questions = questionRepository.findByDifficulty(difficulty);
         if (questions.isEmpty()) {
             throw new QuestionNotFoundException("No questions found with difficulty: " + difficulty);
         }
-        return questions;
+        return questions.stream().map(this::convertToDTO).collect(Collectors.toList());
     }
 
-    public List<Question> getQuestionsByCategory(String category) {
+    public List<QuestionDto> getQuestionsByCategory(String category) {
         List<Question> questions = questionRepository.findByCategoriesContaining(category);
         if (questions.isEmpty()) {
             throw new QuestionNotFoundException("No questions found with category: " + category);
         }
-        return questions;
+        return questions.stream().map(this::convertToDTO).collect(Collectors.toList());
     }
 
-    public Question createQuestion(Question question) {
-        validateQuestion(question);
+    public QuestionDto createQuestion(QuestionDto questionDto) {
+        validateQuestion(questionDto);
 
-        Question newQuestion = new Question();
+        Question newQuestion = convertToDocument(questionDto);
         newQuestion.setId(generateSequence(Question.SEQUENCE_NAME));
-        newQuestion.setTitle(question.getTitle());
-        newQuestion.setDescription(question.getDescription());
-        newQuestion.setDifficulty(question.getDifficulty());
-        newQuestion.setCategories(question.getCategories());
-        newQuestion.setExamples(question.getExamples());
-        newQuestion.setConstraints(question.getConstraints());
-        newQuestion.setLink(question.getLink());
-
-        return questionRepository.save(newQuestion);
+
+        Question createdQuestion = questionRepository.save(newQuestion);
```

Comment:
Suggestion: extract this generateSequence to a separate utility class - it's a generic sequence increment that's not specific to the question.

(but since we only have 1 model here, it's not necessary, just a suggestion)

Reply:
Thanks for the suggestion! I kept it there as what you have mentioned, there's only one model at this point of time but would definitely explore if there are more entities that needs to be handled within question service.